lib/builders/dataTypes/tiptapBuilder/tiptapToolbarBuilder.ts

```diff
@@ -0,0 +1,159 @@
+﻿import {TiptapDataTypeBuilder} from "../tiptapDataTypeBuilder";
+
+export class TiptapToolbarBuilder {
+  parentBuilder: TiptapDataTypeBuilder;
+  toolbarValues: {[key: string]: {value: boolean; row: number}};
+
+  constructor(parentBuilder: TiptapDataTypeBuilder) {
+    this.parentBuilder = parentBuilder;
+    this.toolbarValues = {};
+  }
+
+  withSourceEditor(value: boolean, row: number) {
```

Comment:
Instead of using the row number, I would add another step, in the TipTapBuilder, 

`addToolbarGroup() {
    const builder = new TipTapToolbarGroupBuilder(this);
    this.tipTapToolbarGroupBuilder= builder;
    return builder;
  }`

And in the TipTapToolbarGroupBuilder we would then have AddToolbar.

This would allow us to avoid the row numbers and give us a better overview of the setup of TipTap.